From 0f8358c603c6cc6d6bec4dfcebd6cdd57dfac91b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 31 Oct 2018 11:24:20 +0800
Subject: [PATCH 4968/5242] MLK-20137-1 ARM64: dts: imx8mm-evk-revb: add
 support for panel RM67191

commit  2e118a28a3ae6a7e7b3ed26675395a1192f43ec1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Create a new dts 'fsl-imx8mm-evk-revb-rm67191.dts' to support
panel 'RM67191' display which is attached to DSIM controller
directly on IMX8MM LPDDR4 EVK RevC board to avoid conflict
with ADV7535 display.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../dts/freescale/fsl-imx8mm-evk-revb-rm67191.dts  |   36 ++++++++++++++++++++
 2 files changed, 38 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb-rm67191.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 998cbd8..d4ecc19 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -123,4 +123,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MM) += fsl-imx8mm-evk.dtb \
 				 fsl-imx8mm-evk-rm67191.dtb \
 				 fsl-imx8mm-evk-root.dtb \
 				 fsl-imx8mm-evk-inmate.dtb \
-				 fsl-imx8mm-evk-revb.dtb
+				 fsl-imx8mm-evk-revb.dtb \
+				 fsl-imx8mm-evk-revb-rm67191.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb-rm67191.dts
new file mode 100644
index 0000000..986d2e2
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-revb-rm67191.dts
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mm-evk-revb.dts"
+
+&adv_bridge {
+	status = "disabled";
+};
+
+&mipi_dsi {
+	panel@0 {
+		compatible = "raydium,rm67191";
+		reg = <0>;
+		pinctrl-0 = <&pinctrl_mipi_dsi_en>;
+		reset-gpio = <&gpio1 8 GPIO_ACTIVE_HIGH>;
+		dsi-lanes = <4>;
+		video-mode = <2>;	/* 0: burst mode
+					 * 1: non-burst mode with sync event
+					 * 2: non-burst mode with sync pulse
+					 */
+		panel-width-mm = <68>;
+		panel-height-mm = <121>;
+		status = "okay";
+	};
+};
-- 
1.7.9.5

